fog/view/server/src/block_tracker.rs

```diff
@@ -90,61 +109,49 @@ impl BlockTracker {
                 next_block_count,
             );
 
-            // If the block has been reported as missing, we can advance since we don't need to do
-            // any processing for it.
-            if Self::is_missing_block(missing_block_ranges, next_block_index) {
+            // If the next block index we are checking doesn't exist yet, then we definitely
+            // can't advance the highest processed block count.
+            // This breaks the loop if ingress_keys set is empty.
+            if highest_known_block_count < next_block_count {
```

Comment:
I believe `highest_known_block_count` does not advance when there are missing ranges/gaps. What happens if there's a gap, e.g. key 1 processed blocks 0-100 and then crashed. the network then did 100 blocks with no key being active. key 2 only started at block 200 because it took us time to restart ingest for whatever reason. wouldn't this cause the view server to never advance past 100 since no key provides 101?

Reply:
i think i see what you are saying

in an earlier version, i was using rec.last_scanned_block to decide the highest_known_block_count, then i cut that because there were comments in other tests that that might cause problems and i wasn't sure it was needed

but i think if we don't use that at least for highest_processed_block_count, it's wrong for the reason you say,
and probably it should use that in next_blocks, for the optimization that you mentioned in a different comment

going to think about this, and start by writing tests that would trigger the bad scenario and confirm that it is an issue. then make this change and confirm that it fixes the test if that is indeed the case.